src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java

```diff
@@ -57,6 +57,34 @@
  * <pre>
  * &lt;module name=&quot;InnerAssignment"/&gt;
  * </pre>
+ * <pre>
+ * int a, b, c;
+ * a = b = c = 5; // violation; assignment to each variable should be in a separate statement
+ *
+ * a = 5; // OK
+ * b = 5; // OK
+ * c = 5; // OK
+ * a = 5; b = 5; c = 5; // OK
+ *
+ * double myDouble;
+ * double[] doubleArray = new double[] {myDouble = 4.5, 15.5}; // violation
+ * // assignment to 'myDouble' should be in a separate statement
+ *
+ * String nameOne;
+ * String nameTwo = "tom";
+ * List&lt;String&gt; myList = new LinkedList&lt;String&gt;();
+ * myList.add(nameOne = nameTwo); // violation
```

Comment:
no need for variable `nameTwo`, can be replaced with string literal

Reply:
Sure. Removed.